### PR TITLE
Added Trash Functionality

### PR DESCRIPTION
**Pull request by nikhil-sharma-nks**
Changelogs
- Added Move to trash when deleted
- Added restore from trash
- Added Permanently delete from trash
trash

### CODE REVIEW

File: src/components/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -12,11 +12,11 @@ const Sidebar = () => {
...
-    {
-      name: 'Labels',
-      link: '/labels',
-      icon: <i className='fa-solid fa-tag mr-3'></i>,
-    },
+    // {
+    //   name: 'Labels',
+    //   link: '/home',
+    //   icon: <i className='fa-solid fa-tag mr-3'></i>,
+    // },
```
Comment (atul-gairola) [inline src/components/Sidebar/Sidebar.jsx:19]: please remove commented code

File: src/components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const Sidebar = () => {
     },
     {
       name: 'Profile',
-      link: '/profile',
+      link: '/',
```
Comment (atul-gairola) [inline src/components/Sidebar/Sidebar.jsx:32]: it should have a proper link or not at all. Not a good user experience.

**Review by atul-gairola — APPROVED** [review]
Looks Good To Me. Just take care of these suggestions

File: src/api/NoteService/trash.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      },
+    });
+    if (status >= 200 && status <= 300) return trash;
+    else throw new Error('Error In Deleting From Trash');
+  } catch (err) {
+    console.log(err.message);
+  }
+};
+
+export { getAllTrash, addToTrash, restoreFromTrash, deleteFromTrash };
```
Comment (shlok30) [inline src/api/NoteService/trash.js:81]: Easy to read and understand the services.
Nice clean code

**Review by shlok30 — APPROVED** [review]
Looks good to merge

**Pull request merged**